### DISCUSSION TITLE
Updating an atom and going to different route, the atom is undefined there

### DISCUSSION THREAD

**Discussion by nikivdev** [pmndrs/jotai#1727] [General]
```js
              onClick={async () => {
                const ret = await createReturn()
                setCreatedReturn(ret)
                router.push({
                  pathname: "/success-page",
                })
              }}
              ```
              
I have above onClick. the `ret` is confirmed returning a value. The atom is this one:

`  const [createdReturn, setCreatedReturn] = useAtom(CreatedReturn)`

I don't know why but when I go to that different route and try to do:

`  const [createdReturn] = useAtom(CreatedReturn)`

And then use it in useEffect

```js
  useEffect(() => {
    console.log(createdReturn)
  }, [createdReturn])
```

It is always `undefined`. I don't know why. Am I doing something wrong perhaps?

**Comment by dai-shi** [comment]
Can you create a reproduction with https://codesandbox.io/s/react-new ?
